interview-questions.md

```diff
@@ -7,44 +7,91 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
+  Your answer: the foreign key would be Animal ID. It would be part of the sightings model.
 
   Researched answer:
 
+  When you declare that one model belonds to another, you instruct rails to maintain primary-key foreign-key info between the two instances of models. Rails supports 6 kinds of association:
+        belongs_to
+        has_one
+        has_many
+        has_many :through
+        has_one :through
+        has_and_belongs_to_many
+
 
 
 2. Which RESTful routes must always be passed params? Why?
 
-  Your answer:
+  Your answer: show, edit, update and destroy
 
   Researched answer:
 
+  POST, GET (for show & edit), PATCH & PUT (for update), & DELETE
+
+  These are the routes that need params because they require us to identify parameters from the database in order to make sure the info is updated or created correctly.
+
+  https://learn.co/lessons/sinatra-restful-routes-readme#:~:text=A%20RESTful%20route%20is%20a,HTTP%20verb%20and%20the%20URL.
+
+  ALL the Restful Routes are GET POST SHOW PUT PATCH DELETE
+
 
 
 3. Name three rails generator commands. What is created by each?
 
   Your answer:
+  - rails generate uses a template to create things. if you just run 
+  $ rails generate
+  you can see what is available for you to use.
+
+
+ $ rails generate model ModelNameHere
+ $ rails generate resource 
+ TableNameHere
 
   Researched answer:
 
+  $ rails generate model ModelNameHere
+  $ rails generate resource 
+  $ rails generate controller ControllerName [options]
+    - makes sure a bunch of directories were in our application, and created a controller file, a functional test file, a helper for the view, and a view file.
+  $ rails generate generator
+  $ rails generate scaffold 
+    - this creates a full set of model, database migration for that model, controller to manipulate it, views to view and manipulate the data, and a test suite for each of the above.
+
+
+
+
+
 
 
 4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
-method="GET"    /users          
+method="GET"    /users    index      
 
-method="GET"    /users/1  
+method="GET"    /users/1  show
 
-method="GET"    /users/new
+method="GET"    /users/new    new
 
-method="GET"    /users/edit/1    
+method="GET"    /users/edit/1    create
 
-method="POST"   /users/       
+method="POST"   /users/       edit
 
-method="PUT"    /users/1      
+method="PUT"    /users/1      update
 
-method="DELETE" /users/1      
+method="DELETE" /users/1      destroy
 
 
 
 5. As a developer, you want to make an application that will help you manage your to do list. For the MVP, you want to create just the user's view. Create 10 user stories that will help you get your application started. 
```

Comment:
Great Step by step story I was able to see clearly how you would create the users view. Felt like reading an instructors work. Great job